### PR TITLE
Widgets need to diff with UTC time instead of local time

### LINKED ISSUE #159: The time difference is not calculated based on the time zone
### Dev Home version

_No response_

### Windows build number

_No response_

### Other software

_No response_

### Steps to reproduce the bug

屏幕截图 2023-05-25 093226
It was actually just released, but it shows 8 hours ago (UTC+8)

### Expected result

_No response_

### Actual result

It shows that it was just published or a few seconds ago

### Included System Information

_No response_

### Included Extensions Information

_No response_

### PR DESCRIPTION
**Pull request by EricJohnson327**
## Summary of the pull request
Widgets need to diff with UTC time instead of local time

## PR checklist
- [ ] Closes #159
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Review by guimafelipe — APPROVED** [review]

**Review by guimafelipe — CHANGES_REQUESTED** [review]
This problem also affects the "Mentioned me", "Assigned to me" and "Review requested" widgets. I think it would be better to change the method `DateTimeOffSetDisplayString `in `TimeSpanHelper.cs` and change both Issues and PullRequest widgets to use that method instead.

File: src/GithubPlugin/Widgets/GithubIssuesWidget.cs [hunk 1]
```diff
@@ -132,7 +132,7 @@ public override void LoadContentData()
                     { "title", issueItem.Title },
                     { "url", issueItem.HtmlUrl },
                     { "number", issueItem.Number },
-                    { "date", TimeSpanHelper.TimeSpanToDisplayString(DateTime.Now - issueItem.CreatedAt, Log.Logger()) },
+                    { "date", TimeSpanHelper.TimeSpanToDisplayString(DateTime.UtcNow - issueItem.CreatedAt, Log.Logger()) },
```
Comment (guimafelipe) [inline src/GithubPlugin/Widgets/GithubIssuesWidget.cs:135]:
```suggestion
                    { "date", TimeSpanHelper.DateTimeOffsetToDisplayString(issueItem.CreatedAt, Log.Logger()) },

```

File: src/GithubPlugin/Widgets/GithubPullsWidget.cs [hunk 2]
```diff
@@ -105,7 +105,7 @@ public override void LoadContentData()
                     { "title", pullItem.Title },
                     { "url", pullItem.HtmlUrl },
                     { "number", pullItem.Number },
-                    { "date", TimeSpanHelper.TimeSpanToDisplayString(DateTime.Now - pullItem.CreatedAt, Log.Logger()) },
+                    { "date", TimeSpanHelper.TimeSpanToDisplayString(DateTime.UtcNow - pullItem.CreatedAt, Log.Logger()) },
```
Comment (guimafelipe) [inline src/GithubPlugin/Widgets/GithubPullsWidget.cs:108]:
```suggestion
                    { "date", TimeSpanHelper.DateTimeOffsetToDisplayString(pullItem.CreatedAt, Log.Logger()) },

```

**Review by guimafelipe — CHANGES_REQUESTED** [review]

File: src/GithubPlugin/Widgets/GithubPullsWidget.cs (hunk 2 above)
Comment (EricJohnson327) [reply src/GithubPlugin/Widgets/GithubPullsWidget.cs]: Fixed

File: src/GithubPlugin/Helpers/TimeSpanHelper.cs [hunk 3]
```diff
@@ -2,53 +2,54 @@
 // Licensed under the MIT license.
 using System.Text.Json.Nodes;
 using DevHome.Logging;
-
+using Jeffijoe.MessageFormat;
```
Comment (guimafelipe) [inline src/GithubPlugin/Helpers/TimeSpanHelper.cs:5]: Why is this needed?

**Review by guimafelipe — APPROVED** [review]
LGTM

File: src/GithubPlugin/Helpers/TimeSpanHelper.cs [hunk 4]
```diff
@@ -1,54 +1,54 @@
...
+        }
+
+        if (timeSpan.TotalHours < 2)
+        {
+            return $"1 {Resources.GetResource("WidgetTemplate_HourAgo", log)}";
+        }
+
+        if (timeSpan.TotalDays < 1)
+        {
+            return $"{timeSpan.Hours} {Resources.GetResource("WidgetTemplate_HourssAgo", log)}";
```
Comment (krschau) [inline src/GithubPlugin/Helpers/TimeSpanHelper.cs:43]: We've got a typo in Hours**s**Ago

Comment (EricJohnson327) [inline src/GithubPlugin/Helpers/TimeSpanHelper.cs:43]:
```suggestion
            return $"{timeSpan.Hours} {Resources.GetResource("WidgetTemplate_HoursAgo", log)}";

```

Comment (EricJohnson327) [reply src/GithubPlugin/Helpers/TimeSpanHelper.cs]: Fixed!

File: src/GithubPlugin/Widgets/GithubIssuesWidget.cs [hunk 5]
```diff
@@ -132,7 +132,7 @@ public override void LoadContentData()
                     { "title", issueItem.Title },
                     { "url", issueItem.HtmlUrl },
                     { "number", issueItem.Number },
-                    { "date", TimeSpanHelper.TimeSpanToDisplayString(DateTime.Now - issueItem.CreatedAt, Log.Logger()) },
+                    { "date", TimeSpanHelper.DateTimeOffsetToDisplayString(issueItem.CreatedAt, Log.Logger()) },
```
Comment (krschau) [inline src/GithubPlugin/Widgets/GithubIssuesWidget.cs:135]: I presume issueItem.CreatedAt is in UTC so that the new math works?

Comment (guimafelipe) [reply src/GithubPlugin/Widgets/GithubIssuesWidget.cs:135]: Yes, I tested this here locally and it was showing the right time

**Comment by krschau** [comment]
The hourssago typo is in https://github.com/microsoft/devhomegithubextension/blob/4764c145e04e262cabf63d5ca627e5a4585b1810/src/GithubPlugin/Strings/en-US/Resources.resw#L342 as well

But just those two places.
https://github.com/search?q=repo%3Amicrosoft%2Fdevhomegithubextension+WidgetTemplate_HourssAgo&type=code

File: src/GithubPlugin/Widgets/GithubIssuesWidget.cs (hunk 5 above)
Comment (EricJohnson327) [reply src/GithubPlugin/Widgets/GithubIssuesWidget.cs:135]: @dkbennett said all timestamps in the database are UTC.  So we should be fine here.

**Review by krschau — APPROVED** [review]

File: src/GithubPlugin/Helpers/TimeSpanHelper.cs [hunk 6]
```diff
@@ -1,63 +1,64 @@
...
+
+        if (timeSpan.TotalHours < 2)
+        {
+            return $"1 {Resources.GetResource("WidgetTemplate_HourAgo", log)}";
+        }
+
+        if (timeSpan.TotalDays < 1)
+        {
+            return $"{timeSpan.Hours} {Resources.GetResource("WidgetTemplate_HoursAgo", log)}";
+
```
Comment (guimafelipe) [inline src/GithubPlugin/Helpers/TimeSpanHelper.cs:44]: Extra blank line that breaks the code style.

**Review by guimafelipe — CHANGES_REQUESTED** [review]

**Review by guimafelipe — APPROVED** [review]

**Pull request merged**